[ESLint Plugin] ts-doc-internal checks for hidden instead of ignore

deyaaeldeen opened the pull request:
We are no longer using JSDoc's `@ignore` here https://github.com/Azure/azure-sdk-for-js/pull/12963.

ramya-rao-a commented:
Can we add a note in the md file on when we recommend `@internal` vs `@hidden`? 
Also, am guessing we need to get https://github.com/Azure/azure-sdk-for-js/pull/12963 merged before merging this PR?

deyaaeldeen commented:
@ramya-rao-a @witemple-msft I updated the docs and fixed a bug in the regex used in the rule. Also, https://github.com/Azure/azure-sdk-for-js/pull/12963 has already been merged.
